homeassistant/config.py

```diff
@@ -240,23 +238,45 @@ def set_time_zone(time_zone_str):
             setattr(hac, attr, config[key])
 
     if CONF_TIME_ZONE in config:
-        set_time_zone(config.get(CONF_TIME_ZONE))
+        __set_time_zone(hac, config.get(CONF_TIME_ZONE))
 
     set_customize(config.get(CONF_CUSTOMIZE) or {})
 
     if CONF_TEMPERATURE_UNIT in config:
         hac.temperature_unit = config[CONF_TEMPERATURE_UNIT]
 
+    if CONF_DISTANCE_UNIT in config:
+        hac.distance_unit = config[CONF_DISTANCE_UNIT]
+
+    __auto_discover(hac)
+
+
+def __set_time_zone(hac, time_zone_str):
+    """Helper method to set time zone."""
+    if time_zone_str is None:
+        return
+
+    time_zone = date_util.get_time_zone(time_zone_str)
+
+    if time_zone:
+        hac.time_zone = time_zone
+        date_util.set_default_time_zone(time_zone)
+    else:
+        _LOGGER.error('Received invalid time zone %s', time_zone_str)
+
+
+def __auto_discover(hac):
```

Comment:
Convention in Python is 1 `_` is a private method, 2 `__` is an internal Python method.